src/profile-logic/processed-profile-versioning.js

```diff
@@ -122,6 +122,111 @@ function _mutateProfileToEnsureCauseBacktraces(profile) {
   }
 }
 
+/**
+ * Guess the marker categories for a profile.
+ */
+function _guessMarkerCategories(profile: Object) {
+  // [key, categoryName]
+  const keyToCategoryName = [
+    ['DOMEvent', 'DOM'],
+    ['Navigation::DOMComplete', 'DOM'],
+    ['Navigation::DOMInteractive', 'DOM'],
+    ['Navigation::Start', 'DOM'],
+    ['UserTiming', 'DOM'],
+
+    ['CC', 'GC / CC'],
+    ['GCMajor', 'GC / CC'],
+    ['GCMinor', 'GC / CC'],
+    ['GCSlice', 'GC / CC'],
+
+    ['Paint', 'Graphics'],
+    ['VsyncTimestamp', 'Graphics'],
+    ['CompositorScreenshot', 'Graphics'],
+
+    ['JS allocation', 'JavaScript'],
+
+    ['Styles', 'Layout'],
+    ['nsRefreshDriver::Tick waiting for paint', 'Layout'],
+
+    ['Navigation', 'Network'],
+    ['Network', 'Network'],
+
+    // Explicitly 'Other'
+    ['firstLoadURI', 'Other'],
+    ['IPC', 'Other'],
+    ['Text', 'Other'],
+    ['MainThreadLongTask', 'Other'],
+    ['FileIO', 'Other'],
+    ['Log', 'Other'],
+    ['PreferenceRead', 'Other'],
+    ['BHR-detected hang', 'Other'],
+    ['MainThreadLongTask', 'Other'],
+  ];
+
+  // Make sure the default categories are present since we may want to refer them.
+  for (const defaultCategory of [
+    { name: 'Idle', color: 'transparent', subcategories: ['Other'] },
+    { name: 'Other', color: 'grey', subcategories: ['Other'] },
+    { name: 'Layout', color: 'purple', subcategories: ['Other'] },
+    { name: 'JavaScript', color: 'yellow', subcategories: ['Other'] },
+    { name: 'GC / CC', color: 'orange', subcategories: ['Other'] },
+    { name: 'Network', color: 'lightblue', subcategories: ['Other'] },
+    { name: 'Graphics', color: 'green', subcategories: ['Other'] },
+    { name: 'DOM', color: 'blue', subcategories: ['Other'] },
+  ]) {
+    const index = profile.meta.categories.findIndex(
+      category => category.name === defaultCategory.name
+    );
+    if (index === -1) {
+      // Add on any unknown categories.
+      profile.meta.categories.push(defaultCategory);
+    }
+  }
+
+  const otherCategory = profile.meta.categories.findIndex(
+    category => category.name === 'Other'
+  );
+
+  const keyToCategoryIndex: Map<string, number> = new Map(
+    keyToCategoryName.map(([key, categoryName]) => {
+      const index = profile.meta.categories.findIndex(
+        category => category.name === categoryName
+      );
+      if (index === -1) {
+        throw new Error('Could not find a category index to map to.');
+      }
+      return [key, index];
+    })
+  );
+
+  for (const thread of profile.threads) {
+    const { markers, stringArray } = thread;
+    if (!markers.category) {
+      // Only create the category if it's needed.
+      markers.category = [];
+    }
+    for (let markerIndex = 0; markerIndex < markers.length; markerIndex++) {
+      if (typeof markers.category[markerIndex] === 'number') {
+        // This marker already has a category, skip it.
+        continue;
```

Comment:
Can we change this to `break` instead? It would make the code faster since probably there are either category for every marker or there isn't category for none of them?

Reply:
Sure, works for me.